pkg/3scale/amp/component/system.go

```diff
@@ -506,6 +522,27 @@ func (system *System) appPodVolumes() []v1.Volume {
 	}
 
 	res = append(res, systemConfigVolume)
+
+	if system.Options.S3FileStorageOptions != nil && system.Options.S3FileStorageOptions.S3STSEnabled {
+		s3CredsProjectedVolume := v1.Volume{
+			Name: S3StsCredentialsSecretName,
+			VolumeSource: v1.VolumeSource{
+				Projected: &v1.ProjectedVolumeSource{
+					Sources: []v1.VolumeProjection{
+						v1.VolumeProjection{
+							ServiceAccountToken: &v1.ServiceAccountTokenProjection{
+								Audience:          "openshift",
```

Comment:
This value is also read from the secret, right? Maybe I am mistaken.. this `audience` value neeeds to match the value of `/var/run/secrets/` + `openshift` + `/serviceaccount/token` ? or they are unrelated?

Reply:
I'm not about this either to be honest. I don't think this is related to the secret and has to be certain values configured in the OIDC provider 🤔 

See "Verify that the audience is correct" in https://aws.amazon.com/premiumsupport/knowledge-center/eks-troubleshoot-oidc-and-irsa/

So for Valerys cluster on the associated oidc provider returned:
```
    "ClientIDList": [
        "openshift",
        "sts.amazonaws.com"
    ],
```
So I believe it must be either of these 2 in order to work correctly 🤔

Reply:
maybe we need to add a new field for that in the APIManager CRD???
```yaml
apiVersion: apps.3scale.net/v1alpha1
kind: APIManager
metadata: 
  name: apimanager-sample
  namespace: 3scale-test
spec: 
  system: 
    fileStorage: 
      simpleStorageService: 
        configurationSecretRef: 
          name: s3-credentials
        sts: true
        sts-audience: openshift
```
In that case, we should define a default value or make it mandatory when `sts: true`

Reply:
I think adding that would definitely make the implementation more re-usable for alternative oidc providers. although in the case of 3scale-operator I would anticipate that it's going to be running in openshift mostly anyways. 

In that case I would suggest the following. 
```
spec: 
  system: 
    fileStorage: 
      simpleStorageService: 
        configurationSecretRef: 
          name: s3-credentials
        sts: 
           enabled: true
           audience: openshift
```

Reply:
Yeah, I think adding a new field in the CRD and would defaults to `openshift` if not provided would be good 👍 

This would uncouple the implementation from ROSA STS specifically and allow the user define their own audience if necessary. Though I agree with laura that I'd anticipate this would be mostly running on openshift anyways 🤔 

+1 on the suggested CRD format by @laurafitzgerald